bundles/org.openhab.binding.surepetcare/src/main/java/org/openhab/binding/surepetcare/internal/handler/SurePetcareDeviceHandler.java

```diff
@@ -0,0 +1,254 @@
+                            updateState(DEVICE_CHANNEL_BOWLS_TARGET,
+                                    new QuantityType<Mass>(bowlSettings.targetId, SIUnits.GRAM));
+                        } else if (device.control.bowls.bowlId.equals(4)) {
+                            if ((i + 1) == 1) {
+                                updateState(DEVICE_CHANNEL_BOWLS_FOOD_LEFT,
+                                        new StringType(bowlSettings.foodId.toString()));
+                                updateState(DEVICE_CHANNEL_BOWLS_TARGET_LEFT,
+                                        new QuantityType<Mass>(bowlSettings.targetId, SIUnits.GRAM));
+                            }
+                            if ((i + 1) == 2) {
+                                updateState(DEVICE_CHANNEL_BOWLS_FOOD_RIGHT,
+                                        new StringType(bowlSettings.foodId.toString()));
+                                updateState(DEVICE_CHANNEL_BOWLS_TARGET_RIGHT,
+                                        new QuantityType<Mass>(bowlSettings.targetId, SIUnits.GRAM));
+                            }
+                        }
+                    }
+                    updateState(DEVICE_CHANNEL_BOWLS, new StringType(device.control.bowls.bowlId.toString()));
+                    updateState(DEVICE_CHANNEL_BOWLS_CLOSE_DELAY,
+                            new StringType(device.control.lid.closeDelayId.toString()));
+                    updateState(DEVICE_CHANNEL_BOWLS_TRAINING_MODE,
+                            new StringType(device.control.trainingModeId.toString()));
+                } else {
+                    logger.warn("Unknown product type for device {}", thing.getUID().getAsString());
+                }
+            }
+        }
+    }
+
+    private void updateThingCurfews(SurePetcareDevice device) {
+        for (int i = 0; i < FLAP_MAX_NUMBER_OF_CURFEWS; i++) {
+            SurePetcareDeviceCurfew curfew = device.control.curfewList.get(i);
+
+            updateState(DEVICE_CHANNEL_CURFEW_ENABLED + (i + 1), OnOffType.from(curfew.enabled));
+            updateState(DEVICE_CHANNEL_CURFEW_LOCK_TIME + (i + 1), new StringType(curfew.lockTime));
+            updateState(DEVICE_CHANNEL_CURFEW_UNLOCK_TIME + (i + 1), new StringType(curfew.unlockTime));
+        }
+
+    }
+
+    private void handleCurfewCommand(ChannelUID channelUID, Command command) {
+        String channelUIDBase = channelUID.getIdWithoutGroup().substring(0,
+                channelUID.getIdWithoutGroup().length() - 1);
+        int slot = Integer.parseInt(channelUID.getAsString().substring(channelUID.getAsString().length() - 1));
+
+        synchronized (petcareAPI) {
+            boolean requiresUpdate = false;
+            SurePetcareDevice device = petcareAPI.getDevice(thing.getUID().getId());
+            if (device != null) {
+                SurePetcareDeviceControl existingControl = device.control;
+                logger.debug("Old Curfew Control: {}", existingControl);
```

Comment:
Please take a look at your debug logs a lot of them seem without much context and specific to a debugging session it would be nice to make them more future proof.